src/test/ui/linkage-attr/issue-71519.rs

```diff
@@ -0,0 +1,9 @@
+// test linking using cc with rust-lld injected into search path as ld
+// see rust-lang/rust#71519 for more info
+
+// build-pass
+// only-x86_64-unknown-linux-gnu
+// compile-flags: -C linker-flavor=gcc-lld
+// needs-rust-lld
```

Comment:
I don't think this test will fail when lld is not found.

Reply:
I ignored the test in `src/tools/compiletest/src/header.rs` if rust-lld couldn't be found. I'm still not really satisfied with the test, however, since the test could link using another linker and still pass. I'm not sure how I could check the output as you mentioned [here](https://rust-lang.zulipchat.com/#narrow/stream/182449-t-compiler.2Fhelp/topic/How.20to.20check.20linker.20used.20by.20cc) in a test. Also, I wonder if my test is too trivial. Would there be any weird interactions with larger examples involving multiple crates, etc.?

Reply:
I mean if rust-lld exists and thus the test gets run, but gcc doesn't find/use it for some reason this test will still pass.

Reply:
How would you suggest testing it? It seems like this is not something someone would usually encounter, since the output of linkers are designed to be interchangeable. Also, I couldn't figure out how to extract the output from `-Clink-arg=-Wl,-v` when I entered it in compile-flags, since it seems like the output only gets captured if the test fails.

Reply:
You could add a `run-make` test. That gives you the ability to run arbitrary commands, including checking the rustc output. You could write the output of rustc to a file and then use `$(CGREP)` to check if the output contains a given string. See for example `src/test/run-make/env-dep-info/Makefile` on how to use `$(CGREP)`.